system_metrics_collector/src/topic_statistics_collector/subscriber_topic_statistics.hpp

```diff
@@ -105,7 +106,12 @@ class SubscriberTopicStatisticsNode : public system_metrics_collector::MetricsMe
       };
 
     // Create a publisher with QoS histor_depth set to 10.
-    subscription_ = create_subscription<T>(collect_topic_name_, 10, callback);
+    for (auto & topic_name_str : collect_topic_names_) {
```

Comment:
```suggestion
    for (const auto & topic_name_str : collect_topic_names_) {
```

In C++ everything which can be const should be const. This prevents accidental modifications of data structures.